drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -129,19 +106,13 @@ int netdev2_ieee802154_get(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
             break;
 #ifdef MODULE_GNRC
         case NETOPT_PROTO:
-            if (max_len < sizeof(gnrc_nettype_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(gnrc_nettype_t));
             *((gnrc_nettype_t *)value) = dev->proto;
             res = sizeof(gnrc_nettype_t);
             break;
 #endif
         case NETOPT_DEVICE_TYPE:
-            if (max_len < sizeof(uint16_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(uint16_t));
```

Comment:
Well it is needed, since otherwise the lines below will produce garbage. As to where they get that from: the type of the options is (or for some rare cases should be) well documented in `netopt`. That's how they know. ;)